### PR TITLE
Wip/hughsie/fu device get root

### PR DESCRIPTION
**Pull request by hughsie**

### CODE REVIEW

File: libfwupdplugin/fu-device.c [hunk 1]
```diff
@@ -456,6 +456,31 @@ fu_device_get_parent (FuDevice *self)
 	return priv->parent;
 }
 
+/**
+ * fu_device_get_root:
```
Comment (superm1) [inline libfwupdplugin/fu-device.c]: get root makes me think this is returning the parent of a parent (like the very top chain), but it's just returning the parent of the device.

Comment (hughsie) [reply libfwupdplugin/fu-device.c]: It really is getting the parent of the parent.

File: plugins/vli/fu-vli-usbhub-i2c-device.c [hunk 2]
```diff
@@ -246,6 +247,16 @@ fu_vli_usbhub_i2c_device_write_firmware (FuDevice *device,
 		fu_device_set_progress_full (device, (gsize) j, (gsize) records->len);
 	}
 
+	/* the device automatically reboots */
+	fu_device_set_status (device, FWUPD_STATUS_DEVICE_RESTART);
+	fu_device_set_progress (device, 0);
+
+	/* this is unusual, but the MSP device reboot takes down the entire hub
+	 * for ~60 seconds and we don't want the parent device removing us */
+	root = fu_device_get_root (device);
```
Comment (superm1) [inline plugins/vli/fu-vli-usbhub-i2c-device.c]: do you maybe want an assertion that root != self at the top of this function?

Comment (hughsie) [reply plugins/vli/fu-vli-usbhub-i2c-device.c]: Not sure. I don;t think it's ever possible.

Comment (superm1) [reply plugins/vli/fu-vli-usbhub-i2c-device.c]: If parent wasn't set somehow, that seems like a bad programming error, which is why I thought an assertion made sense.

**Comment by hughsie** [comment]
Self tests FTW.

**Pull request merged**